state-chain/chains/src/btc.rs

```diff
@@ -112,6 +112,24 @@ impl Default for BitcoinTrackedData {
 	}
 }
 
+impl FeeEstimationApi<Bitcoin> for BitcoinTrackedData {
+	fn estimate_ingress_fee(
+		&self,
+		_asset: <Bitcoin as Chain>::ChainAsset,
+	) -> <Bitcoin as Chain>::ChainAmount {
+		// Include the min fee so we over-estimate the cost.
+		self.btc_fee_info.min_fee_required_per_tx + self.btc_fee_info.fee_per_input_utxo
```

Comment:
min_fee_required_per_tx is not an overestimate as we assume 10 bytes, but the maximum size of a transaction that isn't associated with an input/output, seems to be 26 bytes (https://en.bitcoin.it/wiki/Transaction). So we could under-estimate if there are large numbers of inputs or outputs.

Reply:
I'm not clear on the reason a fee numbers to be taken on ingress here. I think a comment would be help as the reasoning for the fees being needed is different for each chain.

In the Eth case, we take fees in the ingress case because we do a fetch into the vault.

But in the Btc case, we take fees to pay for the cost of including the UXTO, in some future egress.

Reply:
As discussed, I agree that we are double-counting the min_fee_required_per_tx but this is deliberate to ensure we overestimate the fees. I agree there is plenty of room for improvement. I'll add some comments that hopefully make some of this a bit clearer.

Reply:
I will add these comments separately if that's ok - have them prepared but forgot to push the commit. And I don't want to push it now and delay the PR anoother 40 minutes while CI completes.